File: clients/client-kotlin/src/jvmTest/kotlin/HasherTest.kt
import org.junit.Test
import kotlin.test.*

//@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class HasherTest {

    private val uuid = "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a";

    private val nonWorkingSeed = 0.22;
    private val workingSeed = 0.11;

    @Test
    fun `should get a value under 40 for a working seed`() {
        assertTrue(Hasher(this.workingSeed).hashString(this.uuid) <= 40);
    }

    @Test
    fun `should get a value above 40 for a non working seed`() {
        assertTrue(Hasher(this.nonWorkingSeed).hashString(this.uuid) > 40)
    }
}

File: clients/csharp/Vexilla.Client.Tests/HasherTest.cs
using NUnit.Framework;

namespace Vexilla.Client.Tests
{
    [TestFixture]
    public class HasherTest
    {
        private const double NonWorkingSeed = 0.22;
        private const string Uuid = "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a";
        private const double WorkingSeed = 0.32;

        [Test]
        public void TestWorkingGradual()
        {
            var result = VexillaHasher.HashString(Uuid, WorkingSeed);
            Assert.True(result <= 0.4);
        }

        [Test]
        public void TestNonWorkingGradual()
        {
            var result = VexillaHasher.HashString(Uuid, NonWorkingSeed);
            Assert.True(result >= 0.4);
        }
    }
}
